enhancements/external-secrets-operator/external-secrets-operator.md

```diff
@@ -0,0 +1,549 @@
+type CertManagerConfig struct {
+	// enabled is for enabling the use of cert-manager for obtaining and renewing the
+	// certificates used for webhook server, instead of built-in certificates.
+	// Use `true` or `false` to indicate the preference.
+	// +kubebuilder:default:="false"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	Enabled string `json:"enabled,omitempty"`
+
+	// addInjectorAnnotations is for adding the `cert-manager.io/inject-ca-from` annotation to the
+	// webhooks and CRDs to automatically setup webhook to the cert-manager CA. This requires
+	// CA Injector to be enabled in cert-manager. Use `true` or `false` to indicate the preference.
+	// +kubebuilder:default:="false"
+	// +kubebuilder:validation:Enum:="true";"false"
+	// +kubebuilder:validation:Optional
+	AddInjectorAnnotations string `json:"addInjectorAnnotations,omitempty"`
+
+	// issuerRef contains details to the referenced object used for
+	// obtaining the certificates. When issuerRef.Kind is Issuer, it must exist in the
+	// .spec.istioCSRConfig.istio.namespace.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="issuerRef is immutable once set"
+	// +kubebuilder:validation:XValidation:rule="self.kind.lowerAscii() == 'issuer' || self.kind.lowerAscii() == 'clusterissuer'",message="kind must be either 'Issuer' or 'ClusterIssuer'"
+	// +kubebuilder:validation:XValidation:rule="self.group.lowerAscii() == 'cert-manager.io'",message="group must be 'cert-manager.io'"
+	// +kubebuilder:validation:Required
+	// +required
+	IssuerRef certmanagerv1.ObjectReference `json:"issuerRef,omitempty"`
+
+	// certificateDuration is the validity period of the webhook certificate.
+	// +kubebuilder:default:="8760h"
+	// +kubebuilder:validation:Optional
+	CertificateDuration *metav1.Duration `json:"certificateDuration,omitempty"`
+
+	// certificateRenewBefore is the ahead time to renew the webhook certificate
+	// before expiry.
+	// +kubebuilder:validation:Optional
+	CertificateRenewBefore *metav1.Duration `json:"certificateRenewBefore,omitempty"`
+}
+```
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+None
+
+#### Standalone Clusters
+
+None
+
+#### Single-node Deployments or MicroShift
+
+None
+
+### Implementation Details/Notes/Constraints
+
+The implementation will make use of static manifests of below kinds to install `external-secrets` and keep the deployment in desired state.
+- Certificates
+- ClusterRoles
+- ClusterRoleBindings
+- Deployments
+- Roles
+- RoleBindings
+- Secrets
+- ServiceAccounts
+- ValidatingWebhookConfigurations
+
+The static manifests are derived from the charts found [here](https://github.com/openshift/external-secrets/tree/main/deploy/charts/external-secrets).
+
+### Risks and Mitigations
+
+The user must keep the `SecretStore` and `ClusterSecretStore` content updated to enable smooth reconciliation of secrets from the providers, 
```

Comment:
Secret zero problem is inherent, a mitigation plan will required for GA.

Reply:
suggestion incorporated.